### PR TITLE
#1825 Patient DOB Picker

### LINKED ISSUE #1825: Dispensing Patient D.O.B should use a date picker
## Describe the bug

At the moment, date of birth displayed in the patient modal is a `toString()` of the `dob` in the db. So it displays as "wed 9th .." etc. But, we need for validation a dob in the form `xx/xx/xxxx`. And, imo, a date picker is far better than typing!

### To reproduce

Development bug as described above

### Expected behaviour

N/A

### Proposed Solution

N/A

### Version and device info

N/A

### Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #1825 

## Change summary

- Adds a date picker for entering date of births and corrects the validation

## Testing

N/A

### Related areas to think about

N/A

### CODE REVIEW

File: src/widgets/FormDateInput.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  // Do not restrict input, but provide feedback to the user.
+  const onChangeTextCallback = React.useCallback(
+    newValue => {
+      const newValueIsValid = onValidate ? onValidate(newValue) : true;
+      onUpdate(newValue, newValueIsValid);
+    },
+    [onValidate]
+  );
+
+  const onChangeDatez = ({ nativeEvent }) => {
```
Comment (mark-prins) [inline src/widgets/FormDateInput.js]: wait, who wrote this? *Datez*?

File: src/widgets/FormDateInput.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    const { timestamp } = nativeEvent;
+    if (!timestamp) return;
+    const newDate = moment(new Date(timestamp)).format('DD/MM/YYYY');
+    onUpdate(newDate, true);
+  };
+
+  const openDatePicker = () => setInputState(state => ({ ...state, datePickerOpen: true }));
+
+  return (
+    <FlexRow flex={1}>
```
Comment (mark-prins) [inline src/widgets/FormDateInput.js]: Is there a reason for `{1}` and not `"1"`? isn't that ever so slightly more expensive to render than a plain string?

Comment (joshxg) [reply src/widgets/FormDateInput.js]: I'm not sure what you mean - I'm guessing you're thinking `{1}` is a JS object? It's just JSX syntax for the parser. If we want a litteral object we need to `{{}}`.. Unless there's something else? Also `flex` needs a number, not a string

File: src/widgets/FormDateInput.js (hunk 1 above)
Comment (joshxg) [reply src/widgets/FormDateInput.js]: I think it was chris !

4d34304

**Pull request merged**